### ISSUE TITLE
TypeError on 5.7.0

### ISSUE THREAD

**Issue by rusefendi** [survivejs/webpack-merge#165]
Node version: 14.15.1
OS: Mac OS 11

This code worked fine on 5.6.1, however on 5.7.0 I am having errors.

```javascript
const base = {
      module: {
        rules: [
          {
            test: /\.s(a|c)ss$/,
            use: [
              {
                loader: "css-loader",
              },
              {
                loader: "sass-loader",
              },
            ],
          },
          {
            test: /\.(png|jpe?g|gif|svg)$/i,
            use: [
              {
                loader: "file-loader",
              },
            ],
          },
        ],
      },
    };
    const development = {
      module: {
        rules: [
          {
            test: /\.s(a|c)ss$/,
            use: ["style-loader"],
          },
        ],
      },
    };

mergeWithRules({
        module: {
          rules: {
            test: "match",
            use: "prepend",
          },
        },
      })(base, development)
```

Error message:     TypeError: Cannot read property 'use' of undefined

**Comment by bebraw** [comment]
Thanks for the report. It feels like a missing piece of logic (likely a missing test earlier). I'll check this one.

**Comment by bebraw** [comment]
Fixed in 5.7.1.
